mllib/src/main/scala/org/apache/spark/ml/feature/ChiSqSelector.scala

```diff
@@ -91,8 +137,17 @@ final class ChiSqSelector @Since("1.6.0") (@Since("1.6.0") override val uid: Str
         case Row(label: Double, features: Vector) =>
           OldLabeledPoint(label, OldVectors.fromML(features))
       }
-    val chiSqSelector = new feature.ChiSqSelector($(numTopFeatures)).fit(input)
-    copyValues(new ChiSqSelectorModel(uid, chiSqSelector).setParent(this))
+    var model = ChiSqSelectorType.withName($(selectorType)) match {
```

Comment:
Sorry, I misunderstand your last comment.  Just think about reuse selector. 
This will not be inconsistent with other methods. I will update the code, thanks.